app/src/main/java/org/simple/clinic/ViewControllerBinding.kt

```diff
@@ -7,19 +7,32 @@ import io.reactivex.android.schedulers.AndroidSchedulers.mainThread
 import io.reactivex.schedulers.Schedulers.io
 import org.simple.clinic.widgets.ScreenDestroyed
 import org.simple.clinic.widgets.UiEvent
+import org.threeten.bp.Duration
+import java.util.concurrent.TimeUnit
 
 @SuppressLint("CheckResult")
 fun <T> bindUiToController(
     ui: T,
     events: Observable<UiEvent>,
     controller: ObservableTransformer<UiEvent, (T) -> Unit>,
-    screenDestroys: Observable<ScreenDestroyed>
+    screenDestroys: Observable<ScreenDestroyed>,
+    uiChangeDelay: Duration = SCREEN_CHANGE_ANIMATION_DURATION
 ) {
   events
       .mergeWith(screenDestroys)
       .observeOn(io())
       .compose(controller)
+      .compose(delayEmissionsBy(uiChangeDelay))
       .observeOn(mainThread())
       .takeUntil(screenDestroys)
       .subscribe { uiChange -> uiChange(ui) }
 }
+
+private fun <T> delayEmissionsBy(duration: Duration): ObservableTransformer<(T) -> Unit, (T) -> Unit> {
+  return ObservableTransformer { upstream ->
+    when (duration) {
+      Duration.ZERO -> upstream
```

Comment:
This statement basically means that we do not want to delay emissions at all.
I was thinking instead of using one method to bind Ui, we could separate this out into two.

Something like this ->

```
fun <T> bindUiToController(
    ui: T,
    events: Observable<UiEvent>,
    controller: ObservableTransformer<UiEvent, (T) -> Unit>,
    screenDestroys: Observable<ScreenDestroyed>
) {
  events
      .mergeWith(screenDestroys)
      .observeOn(io())
      .compose(controller)
      .observeOn(mainThread())
      .takeUntil(screenDestroys)
      .subscribe { uiChange -> uiChange(ui) }
}

fun <T> bindUiToControllerWithDelay(
    ui: T,
    events: Observable<UiEvent>,
    controller: ObservableTransformer<UiEvent, (T) -> Unit>,
    screenDestroys: Observable<ScreenDestroyed>,
    uiChangeDelay: Duration = SCREEN_CHANGE_ANIMATION_DURATION
) {
  events
      .mergeWith(screenDestroys)
      .observeOn(io())
      .compose(controller)
      .delaySubscription(uiChangeDelay.toMillis(), TimeUnit.MILLISECONDS)
      .observeOn(mainThread())
      .takeUntil(screenDestroys)
      .subscribe { uiChange -> uiChange(ui) }
}
```

Reply:
Using the above way we won't have to pass the delay value as `ZERO` explicitly in cases where we do not want to add delay.
Wdyt?

Reply:
Good idea, I'll do this except with one change.

For most of our views, we'll want to add the delay so I'll leave the method named as `bindUiToController` and the other method I'll name `bindUiToControllerWithoutDelay`.

Reply:
Done.